Zend/zend_language_parser.y

```diff
@@ -910,6 +910,8 @@ expr:
 			{ $3->attr = ZEND_ARRAY_SYNTAX_LIST; $$ = zend_ast_create(ZEND_AST_ASSIGN, $3, $6); }
 	|	'[' array_pair_list ']' '=' expr
 			{ $2->attr = ZEND_ARRAY_SYNTAX_SHORT; $$ = zend_ast_create(ZEND_AST_ASSIGN, $2, $5); }
+	|	'[' expr ';' expr ']'
+			{ $$ = zend_ast_create(ZEND_AST_ARRAY_FILL, $2, $4); }
```

Comment:
Did you consider simply translating this into a function call?

```
'[' expr ';' expr ']'
    { zval zero;
      ZVAL_LONG(&zero, 0);
      $$ = zend_ast_create(ZEND_AST_CALL, zend_string_init("array_fill", sizeof("array_fill") - 1, 0),
                                           zend_ast_create_list(3, ZEND_AST_ARG_LIST, zend_ast_create_zval(&zero), $4, $2)
                                           );
    }
```

Note, the above is untested and just off the cuff.